docs/addons/writing-addons.md

```diff
@@ -0,0 +1,226 @@
+</div>
+
+Run `yarn storybook` and you should see something similar to:
+
+![Storybook addon initial state](./addon-initial-state.png)
+
+
+### Display story parameter
+
+Next, let’s replace the `MyPanel` component from above to show the parameter.
+
+<!-- prettier-ignore-start -->
+
+<CodeSnippets
+  paths={[
+    'common/storybook-addon-change-panel.js.mdx',
+  ]}
+/>
+
+<!-- prettier-ignore-end -->
+
+The new version is made smarter by [`useParameter`](./addons-api.md#useparameter), which is a [React hook](https://reactjs.org/docs/hooks-intro.html) that updates the parameter value and re-renders the panel every time the story changes.
+
+The [addon API](./addons-api.md) provides hooks like this so all of that communication can happen behind the scenes. That means you can focus on your addon's functionality.
+
+
+### Using the addon with a story
+
+When Storybook was initialized it provided a small set of example stories that can be used to test if the addon is functioning as expected. Change your `stories/Button.stories.js` to the following:
+
+<!-- prettier-ignore-start -->
+
+<CodeSnippets
+  paths={[
+    'react/button-story-with-addon-example.js.mdx',
+  ]}
+/>
+
+<!-- prettier-ignore-end -->
+
+
+After applying the changes to the story your Storybook UI will now yield the following:
+
+
+<video autoPlay muted playsInline loop>
+  <source
+    src="addon-final-stage-optimized.mp4"
+    type="video/mp4"
+  />
+</video>
+
+### Root level register.js
+
+Before publishing the addon, we'll need to make one additional change. In the root directory of the addon create a new file called `register.js` and add the following:
+
+<!-- prettier-ignore-start -->
+
+<CodeSnippets
+  paths={[
+    'common/my-addon-root-level-register.js.mdx',
+  ]}
+/>
+
+<!-- prettier-ignore-end -->
+
+The sole purpose of making this change is to help with the registration process of the addon. By design Storybook looks for either a `preset.js` or a `register.js` file located at the root level. Omitting this file would require additional configuration from the addon consumer.
+
+
+### Packaging and publishing
+
+With the example above we showed how to create a bare-bones addon. Before publishing and distributing your own addon make sure you have met the following requirements:
+
+- A `package.json file` with metadata about the addon.
+- Peer dependencies of `react` and `@storybook/addons`.
+- A `register.js` file at the root level written as an ES5 module.
+- A `src` directory containing the ES6 addon code.
+- A `dist` directory containing transpiled ES5 code on publish.
+- A [GitHub](https://github.com/) account to host your code.
+- A [NPM](https://www.npmjs.com/) account to publish the addon.
+
+For a good template of an addon that meets the requirements above, check out [@storybook/addon-controls](https://npmjs.com/package/@storybook/addon-controls).
```

Comment:
I think we should find a standalone example. `@storybook/addon-controls` lives in the monorepo and relies on the monorepo's build system. It's really hard to figure out how the addon is actually built and published.

Reply:
@winkerVSbecks we could use the one you've built? WDYT? It's a standalone one outside the monorepo.